lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -84,12 +85,13 @@ def store(credit_card, options={})
         add_extra_data(post, credit_card, options)
         add_recurring_contract(post, options)
         add_address(post, options)
-        commit('authorise', post)
+        commit('authorise', post, options)
       end
 
       def verify(credit_card, options={})
         MultiResponse.run(:use_first_response) do |r|
           r.process { authorize(0, credit_card, options) }
+          options[:idempotency_key] = nil
           r.process(:ignore_result) { void(r.authorization, options) }
```

Comment:
We set the idempotency_key to nil here because we've already used it in the `authorize`? I'm guessing we have to do this because the void will fail because of a duplicate key.

Reply:
Yes, it can't have the key already used in the auth. Look at remote test `test_verify_with_idempotency_key` to see how I checked repeating voids on auths that were already canceled. I'm also not as concerned about this as purchase, as it's a $0-amount, and there's not a lot of hurt in repeat-voids. This could be a big benefit, as re-trying verify without idempotency might raise a fraud flag at the bank for hitting the card with multiple authorizes.

Reply:
Makes sense 👍